### DISCUSSION TITLE
Homepage / index page for a sidebar category?

### DISCUSSION THREAD

**Discussion by borekb** [facebook/docusaurus#3772] [Q&A] [answered]
I have a sidebar like this:

```js
module.exports = {
  docs: {
    'Reference': ['doc1', 'doc2', 'doc3'],
  },
};
```

Which produces this:

- Reference
    - Doc 1 (`/reference/doc1`)
    - Doc 2 (`/reference/doc2`)
    - Doc 3 (`/reference/doc3`)

I'd like to make the route `/reference` work as well and probably write some simple content for it, like "Please select one of the topics from the sidebar" or perhaps auto-generate the content from my actual reference docs.

(I don't want to create a route like `/reference/reference-introduction` – that is still nested while I'd prefer to make the "parent" route working. The use case is that the user / visitor deletes the `/doc1` part of the URL and I'd like them to land on a working page.)

How would I go about that?

**Comment by borekb** [comment]
_(My own brainstorming thread, not really an answer.)_

I'm not even sure how I imagine Docusaurus would support this, I just feel that removing parts of URL should not lead to 404s. For example, I think that all these should render something or redirect somewhere:

- `/docs/reference/doc1`
- `/docs/reference`
- `/docs`

Only the first one works, currently. Maybe the other two could redirect to the "closest reasonable page", like:

- `/docs` redirects to → `/docs/intro/welcome`
- `/docs/reference` redirects to → `/docs/reference/doc1`

Another option would be to support some special file like `<project>/docs/reference/_index.md` which would render on the `/docs/reference` route but would not be visible in a sidebar.

I don't know...

**Reply by borekb** [reply]
I've tried in two projects that I generally admire when it comes to their documentation, and they both produce 404s on manually entered parent URLs:

- Next.js:
    - `200` https://nextjs.org/docs/api-reference/cli
    - `404` https://nextjs.org/docs/api-reference
- Apollo:
    - `200` https://www.apollographql.com/docs/devtools/cli/
    - `404` https://www.apollographql.com/docs/devtools/

🤷‍♂️

**Comment by slorber** [comment, marked as answer]
Hi,

You can control the url of each doc with the slug frontmatter, so it is already possible to create:
- doc1.md: slug: /references
- doc2.md: slug: /references/doc2

Not sure it's related, but we also plan to support sidebar categories linking to a doc:
https://github.com/facebook/docusaurus/issues/2643

We don't do the redirect you ask for by default (it would be hard for us to know  where to redirect to), but you can probably add it in userland. 
There's the redirect plugin to redirect one URL to any other URL you want, and you can create content at any URL thanks to slug.
